src/hrs.ml

```diff
@@ -15,7 +15,7 @@ let print_sym : sym pp = fun oc s ->
 
 (** [print_patt oc p] outputs TPDB format corresponding to the pattern [p], to
     the channel [oc]. *)
-let print_term : bool -> term pp = fun lhs ->
+let rec print_term : bool -> term pp = fun lhs ->
```

Comment:
Why not call `pp`? The internal function is recursive. Is there a difference here?